test/run_e2e_test.sh

```diff
@@ -0,0 +1,20 @@
+TAGS=$@
```

Comment:
Where is the hashbang?  And the other setup for a proper cross-platform shell script?  Shouldn't there be:
```
#!/usr/bin/env bash
set -o errexit
set -o nounset
set -o pipefail
```
Please, educate me if I am expecting complexity that is unneeded!

Reply:
Per our conversation in Slack, I am resolving this and the other comments.  Thank you @jrangelramos for taking the time to discuss most of these comments, and for creating the associated design document in Google Docs.  I am looking forward to seeing the final version